ui/components/src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import {
+  Button,
+  Dialog as MuiDialog,
+  DialogActions,
+  DialogContent,
+  DialogContentProps as MuiDialogContentProps,
+  DialogTitle,
+  DialogTitleProps,
+  IconButton,
+} from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+import { combineSx } from '../utils';
+
+export interface DialogHeaderProps extends DialogTitleProps {
+  /**
+   * Callback fired when close button is clicked. If undefined, close button will not appear in header.
+   */
+  onClose?: (e: React.MouseEvent<HTMLElement>) => void;
+}
+
+export interface DialogButtonProps {
```

Comment:
You should probably extend the material ButtonProps here as well, to have all the power of a regular button. 

If you don't want the ``variant`` and ``type`` properties  you can use Omit

```ts
/**
 * following properties are removed from the DialogButtonProps
 */
interface UnwantedButtonProps {
  variant: unknown;
  type: unknown;
}

export interface DialogButtonProps extends Omit<ButtonProps, keyof UnwantedButtonProps> {}
```